admin_guide/pruning_resources.adoc

```diff
@@ -343,3 +344,152 @@ Such tags tend to have just one image in its history, which effectively prevents
+example.
+- The standard image pruner may delete an image manifest but be interrupted before
+it deletes the related blobs.
+- There was a bug in the registry pruner which failed to remove blobs that were
+supposed to be removed. Image objects referencing them were removed and the
+blobs became orphans.
+// Find this BZ
+
+_Hard pruning_ the registry is a separate procedure from basic image pruning
+that allows you to remove orphaned blobs. You should hard prune if you are
+running out of storage space in your OpenShift Container Registry and believe
+you have orphaned blobs.
+
+This should be an infrequent operation and is necessary only when you have
+evidence that significant numbers of new orphans have been created. Otherwise,
+you can perform standard image pruning at regular intervals, for example, once a
+day (depending on your number of images being created).
+
+To hard prune orphaned blobs the registry:
+
+. *Log in* to xref:../cli_reference/get_started_cli.adoc#basic-setup-and-login[the CLI] as a user with an
+xref:../architecture/additional_concepts/authentication.adoc#oauth[access token].
+
+. *(Optional) Run a basic image prune.* This is an optional step which removes
+additional images that are no longer needed. The hard prune does not remove
+images on its own. It only takes care of blobs stored in the registry storage.
+Therefore it is a good idea run this just before the hard prune.
++
+See xref:pruning-images[Pruning Images] for steps.
+
+. *Switch to read-only mode.* The most important thing to note is that registry
+must be switched to the read-only mode. If not running in this mode, any pushes
+happening at the same time as the prune will either:
++
+--
+- fail and cause new orphans, or
+- succeed although the images will not be pullable (because some of the
+referenced blobs were deleted).
+--
++
+Pushes will not succeed until the registry is switched back to read-write mode.
+Therefore, the hard prune must be carefully scheduled.
++
+To switch the registry to read-only mode, run the following:
++
+----
+$ oc env -n default \
+    dc/docker-registry \
+    'REGISTRY_STORAGE_MAINTENANCE_READONLY={"enabled":true}'
+----
+
+. *Add the system:image-pruner role.* The registry pod needs additional
+permissions in order to list some resources. Ensure the service account used to
+run the registry instances has the cluster role *system:image-pruner*.
+
+.. Get the service account name:
++
+----
+$ service_account=$(oc get -n default \
+    -o jsonpath=$'system:serviceaccount:{.metadata.namespace}:{.spec.template.spec.serviceAccountName}\n' dc/docker-registry)
+----
+
+.. Add the cluster role to the service account:
++
+----
+$ oadm policy add-cluster-role-to-user \
+    system:image-pruner \
+    ${service_account)
+----
+
+. *(Optional) Run the pruner in dry-run mode.* To get an idea for what will be
+removed by the hard pruner, it can be run in a dry-run mode where no changes
+are actually made:
```

Comment:
To get a list of blobs that will be removed, run the hard pruner in dry-run mode. No changes are actually made. (??)